OptionSelector/OptionSelectorTableViewController.m

```diff
@@ -0,0 +1,205 @@
+    
+    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:@"OptionSelectorCellIdentifier" forIndexPath:indexPath];
+
+    Type *tmp = self.array[indexPath.row];
+    cell.textLabel.text = tmp.name;
+    
+    if (indexPath.row == 0) {
+        cell.detailTextLabel.text =  [tmp.selection isEqualToString:@" "] ? [[NSUserDefaults standardUserDefaults] objectForKey:@"cell1"] : tmp.selection;
+    }
+    else if(indexPath.row == 1){
+        cell.detailTextLabel.text = [tmp.selection isEqualToString:@" "] ? [[NSUserDefaults standardUserDefaults] objectForKey:@"cell2"]: tmp.selection;
+    }
+    else{
+        cell.detailTextLabel.text =[tmp.selection isEqualToString:@" "] ? [[NSUserDefaults standardUserDefaults] objectForKey:@"cell3"]: tmp.selection;
+    }
+    
+    return cell;
+}
+//
+//-(void)tableView:(UITableView *)tableView willDisplayCell:(UITableViewCell *)cell forRowAtIndexPath:(NSIndexPath *)indexPath{
+//    if(self.selectedData.count != 0){
+//    UITableViewCell *selectedCell = self.selectedData[indexPath.row];
+//    NSString *detail = selectedCell.textLabel.text;
+//        cell.detailTextLabel.text = detail;
+//    }
+//}
+
+/*
+// Override to support conditional editing of the table view.
+- (BOOL)tableView:(UITableView *)tableView canEditRowAtIndexPath:(NSIndexPath *)indexPath {
+    // Return NO if you do not want the specified item to be editable.
+    return YES;
+}
+*/
+
+/*
+// Override to support editing the table view.
+- (void)tableView:(UITableView *)tableView commitEditingStyle:(UITableViewCellEditingStyle)editingStyle forRowAtIndexPath:(NSIndexPath *)indexPath {
+    if (editingStyle == UITableViewCellEditingStyleDelete) {
+        // Delete the row from the data source
+        [tableView deleteRowsAtIndexPaths:@[indexPath] withRowAnimation:UITableViewRowAnimationFade];
+    } else if (editingStyle == UITableViewCellEditingStyleInsert) {
+        // Create a new instance of the appropriate class, insert it into the array, and add a new row to the table view
+    }   
+}
+*/
+
+/*
+// Override to support rearranging the table view.
+- (void)tableView:(UITableView *)tableView moveRowAtIndexPath:(NSIndexPath *)fromIndexPath toIndexPath:(NSIndexPath *)toIndexPath {
+}
+*/
+
+/*
+// Override to support conditional rearranging of the table view.
+- (BOOL)tableView:(UITableView *)tableView canMoveRowAtIndexPath:(NSIndexPath *)indexPath {
+    // Return NO if you do not want the item to be re-orderable.
+    return YES;
+}
+*/
+
+
+#pragma mark - Navigation
+
+// In a storyboard-based application, you will often want to do a little preparation before navigation
+- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
+    
+    OptionSelectorDetailTableViewController *dtvc = [segue destinationViewController];
+    
+    NSIndexPath *indexPath = [self.tableView indexPathForSelectedRow];
+    self.index = indexPath.row;
```

Comment:
If you're going to use a property to store the index of the selected type (which you are using to replace that type in your array later), then I would suggest giving it a more descriptive name. Something like `indexOfCurrentType` is clearer.

Generally, though, I'm not sure that this is the best approach to solving the problem at hand. It introduces a dependency between two methods and adds global state.

I was going to suggest an improvement, but from this comment https://github.com/accesscode-2-2/unit-1-hw-week-4/pull/13/files#diff-12d5250b3d4c51e0042cf1c7053c6a12R115 it looks that you've already figured this out. :smile: So you can just rip out the whole `index` property and the replacing of the object in the array.